File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n=nums.size();
        vector<int> v;
        int count=0;
        for(int i=0; i<n; i++){
            if(nums[i]==0){
                count++;
            }
        }
        for(int i=0; i<n; i++){
            if(nums[i]!=0){
                v.push_back(nums[i]);
            }
        }
        for(int i=0; i<count; i++){
            v.push_back(0);
        }
        nums=v;
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        List<Integer> list=new ArrayList<>();
        for(int x: nums){
            list.add(x);
        }
        int count=0;
        List<Integer> v=new ArrayList<>();
        for(int i=0; i<list.size();i++){
            if(list.get(i)!=0){
                v.add(list.get(i));
            }
            else{
                count++;
            }
        }
        for(int i=0; i<count; i++){
            v.add(0);
        }
        for(int i=0; i<nums.length; i++){
            nums[i]=v.get(i);
        }
    }
}